packages/server/utils/setUserTierForUserIds.ts

```diff
@@ -35,6 +39,34 @@ const setUserTierForUserIds = async (userIds: string[]) => {
       {nonAtomic: true}
     )
     .run()
+
+  const userTiers = (await r
+    .table('OrganizationUser')
+    .getAll(r.args(userIds), {index: 'userId'})
+    .filter({removedAt: null})
+    .merge((orgUser) => ({
+      tier: r
+        .db('actionDevelopment')
+        .table('Organization')
+        .get(orgUser('orgId'))('tier')
+        .default('personal')
+    }))
+    .group('userId')('tier')
+    .ungroup()
+    .merge((row) => ({
+      tier: row('reduction')
+        .reduce((_left, right) => {
+          return r.branch(right.eq('enterprise'), 'enterprise', right.eq('pro'), 'pro', 'personal')
+        })
+        .default('personal')
+    }))
+    .map((row) => ({
+      id: row('group'),
+      tier: row('tier')
+    }))
+    .run()) as {id: string; tier: TierEnum}[]
```

Comment:
i noticed that this query might not be equivalent to the original, which i confirmed by testing locally.

to see the difference:
- login as a new user locally
- create many new teams off of many new orgs, e.g. (org2, team2), (org3, team3), etc...
- go to one of the random orgs and upgrade to pro
- check in rethinkdb that the user row will always have the "highest" tier of any orgs they're a part of
- check in pg and see that the tier will *probably* still be "personal". (it's possible by luck the tier could be "pro" too, but the point being that it's not deterministic)

i think it will be fine to just keep the original code, and only change getting `OrganizationUser`s by directly passing in the `userIds` as you did here. can you make those changes?

Reply:
That's a great find, thank you!!! I think the non-deterministic logic come from this part:

```typescript
row('reduction')
  .reduce((_left, right) => {
    return r.branch(right.eq('enterprise'), 'enterprise', right.eq('pro'), 'pro', 'personal')
  })
  .default('personal')
```

So `['pro', 'personal']` would just yield `'personal'` whereas `['personal', 'pro']` would give you `'pro'`...

Anyway I took your advice and just copied the original logic directly and modified a little bit to make it work with PG.

Reply:
oh no! it looks like i missed something. for the query to get `userTiers`, can you remove the dependency on rethink `User` table? the goal is that we want to be able to drop rethink User table, and we can't do that if any lingering queries use the table. Let's just get `OrganizationUser`s from the passed `userIds` arg using the `OrganizationUser.userId` index (i think this is what you were doing at first), and same thing for the `id` property in the mapped object.

Reply:
You are so right! Guess I was in hurry fixing the bug in the logic and then completely forgot why we needed to rewrite this lol.

Reply:
cool, tested and looks good!